core/src/main/java/io/spine/core/Commands.java

```diff
@@ -225,21 +225,26 @@ public static CommandId checkValid(CommandId id) {
     }
 
     /**
-     * Verifies if the exception has command rejection as its
-     * {@linkplain Throwables#getRootCause(Throwable) root cause} .
+     * Produces a {@link Rejection} for the given {@link Command} based on the given
+     * {@linkplain ThrowableMessage cause}.
      *
-     * @param exception the exception to analyze
-     * @return {@code true} if the exception was created because of a command rejection thrown,
-     *         {@code false} otherwise
+     * <p>The given {@link Throwable} should be
+     * {@linkplain Rejections#causedByRejection caused by a Rejection} or
+     * an {@link IllegalArgumentException} is thrown.
+     *
+     * @param command the command to reject
+     * @param cause the rejection cause (may be wrapped into other kinds of {@code Throwable})
+     * @return a {@link Rejection} for the given command
+     * @throws IllegalArgumentException upon an invalid rejection cause
      */
-    public static boolean causedByRejection(RuntimeException exception) {
-        //TODO:2017-07-26:alexander.yevsyukov: Check against CommandRejection
-        // instead of ThrowableMessage when code generation allows customizing a custom
-        // rejection types instead of `ThrowableMessage`.
-        // See: https://github.com/SpineEventEngine/base/issues/20
-        final Throwable rootCause = Throwables.getRootCause(exception);
-        final boolean result = rootCause instanceof ThrowableMessage;
-        return result;
+    @Internal
+    public static Rejection reject(Command command, Throwable cause) throws IllegalStateException {
```

Comment:
`rejectWithCause`
Also, `IllegalStateException` makes not much sense. In case it's thrown, the _argument_ is wrong. Not the state.